samples/sensor/sensor_shell/src/main.c

```diff
@@ -4,6 +4,64 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <stdlib.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
 #include <zephyr/logging/log.h>
 
 LOG_MODULE_REGISTER(app);
+
+struct sample_stats {
+	int64_t accumulator;
+	uint32_t count;
+	uint64_t sample_window_start;
+};
+
+static void data_ready_trigger_handler(const struct device *sensor,
+				       const struct sensor_trigger *trigger)
+{
+	static struct sample_stats stats[SENSOR_CHAN_ALL];
+	const int64_t now = k_uptime_get();
+	struct sensor_value value;
+
+	if (sensor_sample_fetch(sensor)) {
+		LOG_ERR("Failed to fetch samples on data ready handler");
+	}
+	for (int i = 0; i < SENSOR_CHAN_ALL; ++i) {
+		/* Skip 3 axis channels */
+		if (i == SENSOR_CHAN_ACCEL_XYZ || i == SENSOR_CHAN_GYRO_XYZ ||
+		    i == SENSOR_CHAN_MAGN_XYZ) {
+			continue;
+		}
+		if (sensor_channel_get(sensor, i, &value) != 0) {
```

Comment:
This causes a lot of errors on `frdm_k64f` when you enable the data ready trigger:
```
$ west build -b frdm_k64f -p auto samples/sensor/sensor_shell/ -- -DCONFIG_FXOS8700_TRIGGER_GLOBAL_THREAD=y
```
```console
--- 30 messages dropped ---
[00:00:09.459,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.459,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.459,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
--- 29 messages dropped ---
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
```

Reply:
I updated the logic to try once and if we get `-ENOTSUP` then never try that channel again

Reply:
Right we don't really have a nice way of introspecting supported channels at runtime or compile time at the moment per sensor. Makes sense to me to try once.